src/SpectatorView.Native/SpectatorView.Compositor/Compositor/AzureKinectCameraFrame.cpp

```diff
@@ -0,0 +1,116 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#include "pch.h"
+#include "AzureKinectCameraFrame.h"
+
+AzureKinectCameraFrame::AzureKinectCameraFrame(bool captureDepth, bool captureBodyMask)
+    : _captureDepth(captureDepth)
+    , _captureBodyMask(captureBodyMask)
+    , _status(FrameStatus::Unused)
+{
+    _imageSizes[(int)AzureKinectImageType::Color] = FRAME_BUFSIZE_RGBA;
+    _imageSizes[(int)AzureKinectImageType::Depth] = FRAME_BUFSIZE_DEPTH16;
+    _imageSizes[(int)AzureKinectImageType::BodyMask] = FRAME_BUFSIZE_DEPTH16;
+
+    for (int i = 0; i < AZURE_KINECT_IMAGE_TYPE_COUNT; i++)
+    {
+        _images[i] = new uint8_t[_imageSizes[i]];
+    }
+}
+
+AzureKinectCameraFrame::~AzureKinectCameraFrame()
+{
+    for (int i = 0; i < AZURE_KINECT_IMAGE_TYPE_COUNT; i++)
+    {
+        delete[] _images[i];
+    }
+}
+
+void AzureKinectCameraFrame::StageImage(AzureKinectImageType imageType, k4a_image_t image)
+{
```

Comment:
it seems like there should be some sort of guard around this. Like we should obligate the caller to declare it has the held lock

Reply:
granted, it may just be better to have a try stage where we don't write if the status isn't writing

---
In reply to: [403176779](https://github.com/microsoft/MixedReality-SpectatorView/pull/356#discussion_r403176779) [](ancestors = 403176779)